### ISSUE TITLE
updating table build

### ISSUE THREAD

**Issue by lachlandeer** [lachlandeer/snakemake-econ-r#14]
two things to fix:

1) move from stargazer to modelsummary
2) change table-specs to be json dicts and a subdir of `src` not `tables`

**Comment by lachlandeer** [comment]
Here's code that would replace stargazer output with approx the same

```{r}
library(modelsummary)
library(kableExtra)

models <- list(
    solow_oecd <- lm(ln_gdp_85 ~ ln_inv_gdp + ln_ndg, data = oecd),
    solow_int  <- lm(ln_gdp_85 ~ ln_inv_gdp + ln_ndg, data = intermediate),
    solow_noil <- lm(ln_gdp_85 ~ ln_inv_gdp + ln_ndg, data = non_oil),
    solow_oecd_r <- lm(ln_gdp_85 ~ I(ln_inv_gdp - ln_ndg), data = oecd),
    solow_int_r  <- lm(ln_gdp_85 ~ I(ln_inv_gdp - ln_ndg), data = intermediate),
    solow_noil_r <- lm(ln_gdp_85 ~ I(ln_inv_gdp - ln_ndg), data = non_oil)
)

names(models) <- c("(1)", "(2)", "(3)", "(4)", "(5)", "(6)")

custom_row <- tibble::tribble(~term, ~"(1)", ~"(2)", ~"(3)", ~"(4)", ~"(5)", ~"(6)",
                              "Restricted Model", "No", "No", "No", "Yes", "Yes", "Yes")

attr(custom_row, 'position') <- c(9)

f_int <- function(x) format(round(x, 2), big.mark=",", nsmall = 0) # integers
f_dec <- function(x) format(round(x, 2), big.mark=",", nsmall = 2) # floats

gm <- list(
  list("raw" = "nobs", "clean" = "N", "fmt" = f_int),
  list("raw" = "r.squared", "clean" = "$R^2$", "fmt" = f_dec))

tab <- modelsummary(models,
             fmt = "%.2f",
             coef_map = c(    "ln_inv_gdp" = "log(I / GDP)",
                              "ln_ndg" = "log(n + g + $\\delta$)",
                              "I(ln_inv_gdp - ln_ndg)"=  "log(I / GDP) - log(n + g + $\\delta$)",
                              '(Intercept)' = 'Constant'
                               ),
             statistic = "std.error",
             stars = FALSE,
             stars_note = FALSE,
             gof_omit = 'DF|Deviance|Adj|AIC|BIC|Log.|F',
             gof_map = gm,
             add_rows = custom_row,
             output = 'latex_tabular',
             escape = FALSE # render latex as latex- needs FALSE
             )

tab2 <- tab %>%
    add_header_above(c(" " =1, "Non-Oil" =1, "Intermediate" =1 , "OECD"=1,
                            "Non-Oil"=1, "Intermediate"=1, "OECD"=1)) %>%
    add_header_above(c(" " =1, "Log(GDP '85) - Log(GDP '60)" =6)) %>%
    row_spec(9, extra_latex_after = "\\midrule") %>%
    save_kable(file = "table.tex", self_contained = FALSE)


```

**Comment by lachlandeer** [comment]
fixed in #15
